### PR TITLE
Trajectory generator fixes

### PR DESCRIPTION
**Pull request by bresch**
### Context:
The trajectory generator (called VelocitySmoothing here) is basically an optimizer that tries to find an optimal jerk signal of fixed amplitude to control each output of a chain of two integrators (i.e.: control acceleration and velocity) while minimizing the time to reach the desired state and satisfying maximum acceleration.

### Problem:
When the defined jerk is large and the velocity/acceleration targets are small, the problem cannot be solved given the fixed time interval (this is a discrete-time problem). The drone sometimes keeps drifting while the stick is centered because the velocity is non-zero.

### Proposed solution:
When the velocity and acceleration are small, decrease the jerk (set it to 1) such that the optimizer can find a new solution and move closer to zero. Once the outputs are small enough, force them to 0.

### Other changes:
- Some division by 0 and sqrt of negative number protections.
- A simple script that uses the VelocitySmoothing library. This is used to debug the code, verify the outputs and compare them to the original python script. The next step is to run a true unit test to compare the outputs automatically.

### CODE REVIEW

File: src/lib/FlightTasks/tasks/ManualPositionSmoothVel/FlightTaskManualPositionSmoothVel.cpp [hunk 1]
```diff
@@ -196,10 +208,28 @@ void FlightTaskManualPositionSmoothVel::_updateSetpoints()
 	if (_position_lock_xy_active) {
 		_position_setpoint_xy_locked(0) = pos_sp_smooth(0);
 		_position_setpoint_xy_locked(1) = pos_sp_smooth(1);
+
+		// If the velocity setpoint is smaller than 1mm/s and that the acceleration is 0, force the setpoints
```
Comment (bkueng) [inline src/lib/FlightTasks/tasks/ManualPositionSmoothVel/FlightTaskManualPositionSmoothVel.cpp]: I'd add a comment explaining why we do this.

File: src/lib/FlightTasks/tasks/Utility/VelocitySmoothing.cpp [hunk 2]
```diff
@@ -109,16 +107,23 @@ float VelocitySmoothing::computeT1(float T123, float accel_prev, float vel_prev,
 {
 	float a = -max_jerk;
 	float b = max_jerk * T123 - accel_prev;
-	float delta = T123 * T123 * max_jerk * max_jerk + 2.f * T123 * accel_prev * max_jerk - accel_prev * accel_prev
-		      + 4.f * max_jerk * (vel_prev - vel_setpoint);
+	float delta;
```
Comment (bkueng) [inline src/lib/FlightTasks/tasks/Utility/VelocitySmoothing.cpp]: Why did you separate the assignment?

File: src/lib/FlightTasks/tasks/Utility/VelocitySmoothing.cpp [hunk 3]
```diff
@@ -147,7 +152,13 @@ float VelocitySmoothing::computeT2(float T1, float T3, float accel_prev, float v
 {
 	float f = accel_prev * T1 + max_jerk * T1 * T1 * 0.5f + vel_prev + accel_prev * T3 + max_jerk * T1 * T3
 		  - max_jerk * T3 * T3 * 0.5f;
-	float T2 = (vel_setpoint - f) / (accel_prev + max_jerk * T1);
+	float T2 = 0.f;
+
+	float den = (accel_prev + max_jerk * T1);
```
Comment (bkueng) [inline src/lib/FlightTasks/tasks/Utility/VelocitySmoothing.cpp]:
```suggestion
	float den = accel_prev + max_jerk * T1;
```

**Comment by EliaTarasov** [comment]
@bresch 

> Context:
The trajectory generator (called VelocitySmoothing here) is basically an optimizer that tries to find an optimal jerk signal of fixed amplitude to control each output of a chain of two integrators (i.e.: control acceleration and velocity) while minimizing the time to reach the desired state and satisfying maximum acceleration.

Can you please describe in formulas (or refer document(s)) how this oprimizer exactly works?

**Comment by bresch** [comment]
@bkueng Thanks for the review, I updated the PR.
I agree, this is not the best solution solution, I would have to find a proper way to automatically adjust the jerk. I experimented in my python script to recompute the jerk in function of the current states and objectives, but it also has some corner cases. This solution is light and covers most of the potential issues.

**Pull request merged**